scripts/plotting_tools.py

```diff
@@ -0,0 +1,329 @@
+import os
+from os.path import join
+from glob import glob
+
+import pandas as pd
+import xarray as xr
+import numpy as np
+
+from matplotlib import pyplot as plt
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+
+import logging
+
+ptypes = ['snow', 'rain', 'frzr', 'icep']
+
+def get_tle_files(base_path, valid_time, n_members=1, min_lead_time=1, init_step=1):
+    '''
+    Get ptype files at a given valid time.
+
+    Arguments:
+        base_path (str): path to directory
+        valid_time (str): YYYY-mm-dd HHMM
+        n_members (int): for gathering an ensemble of times, default is no ensemble
+        min_lead_time (int): ?
+        init_step (int): not implemented 
+    '''
+    date = pd.to_datetime(valid_time)
+    file_names = []
+    if n_members == 1:
+        dt = date - pd.Timedelta(f"1h")
+        date_str = f"MILES_ptype_hrrr_{dt.strftime('%Y-%m-%d_%H%M')}"
+        return [join(base_path, dt.strftime("%Y%m%d"), dt.strftime("%H%M"), f"{date_str}_f01.nc")]
+    for time_delta in range(n_members, 0, -1):
+        dt = date - pd.Timedelta(f"{time_delta}h")
+        date_str = f"MILES_ptype_hrrr_{dt.strftime('%Y-%m-%d_%H%M')}"
+        path = join(base_path, dt.strftime("%Y%m%d"), dt.strftime("%H%M"), f"{date_str}_f{str(time_delta).zfill(2)}.nc")
+        file_names.append(path)
+    return file_names
+
+
+def load_data(files, variables):
+    def process(ds):
+        ds = ds[variables]
+        precip_sum = ds[['crain', 'csnow', 'cicep', 'cfrzr']].to_array().sum(dim='variable')
+        for ptype in ptypes:
+            ds[f'ML_c{ptype}'] = xr.where(precip_sum >= 1, x=ds[f"ML_c{ptype}"], y=np.nan)
```

Comment:
Can you add some documentation here? Specifically how the masking works would be useful for future selves and others.